src/avro/src/reader.rs

```diff
@@ -599,15 +599,38 @@ impl<'a> SchemaResolver<'a> {
             // Any other anonymous type.
             (SchemaPieceRefOrNamed::Piece(wp), SchemaPieceRefOrNamed::Piece(rp)) => {
                 match (wp, rp) {
+                    // Normally for types that are underlyingly "long", we just interpret them according to the reader schema.
+                    // In this special case, it is better to interpret them according to the _writer_ schema:
```

Comment:
can you explain why it's better here, instead of just stating that it is?

Reply:
Sure, I will expand on it. Briefly the idea is that if a writer wrote milliseconds and a reader tries to read it as microseconds, it will be off by a factor of 1000 from the timestamp that the writer was intending to write

Reply:
I guess that's because if the writer wrote it as micros and the reader is expecting millis then the reader is just wrong, and the reader shouldn't be able to specify what the unit is, it just knows that it wants a datetime? If correct, I think it's still worth writing that down it took me a sec to figure out why it might be the case here and not other places that are basically primitives.

Reply:
"if the writer wrote it as micros and the reader is expecting millis then the reader is just wrong"

Indeed - but the whole idea behind schema resolution/evolution is to do "the right thing" whenver you can, even if the writer and reader are expecting different things.

The difference between here and the other primitives is there is indeed some semantics to how things are interpreted that we don't have with the primitives.

It's a good point, I'll expand on the comment before landing this.

Reply:
thanks!